File: src/project_manager_flutter/lib/features/dashboard/repository/dashboard_repository.dart
import 'package:dio/dio.dart';
import '../../../core/constants/api_constants.dart';
import '../../../core/network/api_exceptions.dart';
import '../models/dashboard_summary.dart';

class DashboardRepository {
  final Dio _dio;

  DashboardRepository(this._dio);

  Future<DashboardSummary> getSummary() async {
    try {
      final response = await _dio.get(ApiConstants.dashboardSummary);
      return DashboardSummary.fromJson(response.data);
    } on DioException catch (e) {
      throw ApiException.fromResponse(
        e.response?.statusCode ?? 500,
        e.response?.data,
      );
    }
  }
}

File: src/project-manager-web/src/features/dashboard/repository/dashboard-repository.ts
import { apiClient } from "@/core/api/client";
import { ApiEndpoints } from "@/core/api/constants";
import { ApiException } from "@/core/api/exceptions";
import type { DashboardSummary } from "../models/types";
import { AxiosError } from "axios";

export const dashboardRepository = {
  async getSummary(): Promise<DashboardSummary> {
    try {
      const response = await apiClient.get<DashboardSummary>(ApiEndpoints.dashboardSummary);
      return response.data;
    } catch (error) {
      if (error instanceof AxiosError) {
        throw ApiException.fromResponse(error.response?.status ?? 500, error.response?.data);
      }
      throw error;
    }
  },
};
